src/sensors/mpu9250sensor.cpp

```diff
@@ -382,4 +366,101 @@ void MPU9250Sensor::startCalibration(int calibrationType) {
+    Mxyz[1] = (float)data[1];
+    Mxyz[2] = -(float)data[2];
+
+    float temp[3];
+
+    //apply offsets and scale factors from Magneto
+    for (unsigned i = 0; i < 3; i++) {
+        temp[i] = (Mxyz[i] - m_Calibration.M_B[i]);
+        #if useFullCalibrationMatrix == true
+            Mxyz[i] = m_Calibration.M_Ainv[i][0] * temp[0] + m_Calibration.M_Ainv[i][1] * temp[1] + m_Calibration.M_Ainv[i][2] * temp[2];
+        #else
+            Mxyz[i] = temp[i];
+        #endif
+    }
 }
+
+void MPU9250Sensor::parseAccelData(int16_t data[3]) {
+    // reading big endian int16
+    Axyz[0] = (float)data[0];
+    Axyz[1] = (float)data[1];
+    Axyz[2] = (float)data[2];
+
+    float temp[3];
+
+    //apply offsets (bias) and scale factors from Magneto
+    for (unsigned i = 0; i < 3; i++) {
+        temp[i] = (Axyz[i] - m_Calibration.A_B[i]);
+        #if useFullCalibrationMatrix == true
+            Axyz[i] = m_Calibration.A_Ainv[i][0] * temp[0] + m_Calibration.A_Ainv[i][1] * temp[1] + m_Calibration.A_Ainv[i][2] * temp[2];
+        #else
+            Axyz[i] = temp[i];
+        #endif
+    }
+}
+
+// TODO: refactor so that calibration/conversion to float is only done in one place.
+void MPU9250Sensor::parseGyroData(int16_t data[3]) {
+    // reading big endian int16
+    Gxyz[0] = ((float)data[0] - m_Calibration.G_off[0]) * gscale; //250 LSB(d/s) default to radians/s
+    Gxyz[1] = ((float)data[1] - m_Calibration.G_off[1]) * gscale;
+    Gxyz[2] = ((float)data[2] - m_Calibration.G_off[2]) * gscale;
+}
+
+// really just an implementation detail of getNextSample...
+void MPU9250Sensor::swapFifoData(union fifo_sample_raw* sample) {
+    #if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+        // byteswap the big endian integers
+        for (unsigned iii = 0; iii < 12; iii += 2) {
+            uint8_t tmp = sample->raw[iii + 0];
+            sample->raw[iii + 0] = sample->raw[iii + 1];
+            sample->raw[iii + 1] = tmp;
+        }
+    #elif __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+        // byteswap the little endian integers
+        for (unsigned iii = 12; iii < 18; iii += 2) {
+            uint8_t tmp = sample->raw[iii + 0];
+            sample->raw[iii + 0] = sample->raw[iii + 1];
+            sample->raw[iii + 1] = tmp;
+        }
+    #else
+        #error "Endian isn't Little endian or big endian, are we not using GCC or is this a PDP?"
+    #endif
+
+    // compiler hint for the union, should be optimized away for optimization >= -O1 according to compiler explorer
+    memmove(&sample->sample, sample->raw, sensor_data_len);
+}
+
+// thought experiments:
+// is a single burst i2c transaction faster than multiple? by how much?
+// how does that compare to the performance of a memcpy?
+// how does that compare to the performance of data fusion?
+// if we read an extra byte from the magnetometer (or otherwise did something funky)
+// we could read into a properly aligned array of fifo_samples (and not require a memcpy?)
```

Comment:
This thought experiment is asking the same question as "would it be faster to read multiple samples at once" up above, except in more documentation and detail, in case this gets merged and someone is poking around for something to try.

The comment about an extra byte from the magnetometer is referring to the fact that a single sample is currently 19 bytes, because of the magnetometer status byte. If we had an array of `fifo_sample`, there would likely be a single byte of padding between each sample for 16 bit address alignment, because these structures mostly consist of 16 bit integers. If we read an extra byte from the magnetometer, then we could have a sample that is 20 bytes, and requires no additional padding bytes.